### DISCUSSION TITLE
nats-api binary files

### DISCUSSION THREAD

**Discussion by SimpleTechGuy** [amidaware/tacticalrmm#1784] [General]
Hi, hoping someone here might be able to shed a little light on this for me.  I've been looking around a bit and maybe I'm missing something, but can anybody explain the `natsapi/bin/nats-api` and `natsapi/bin/nats-api-arm64` binary files. It appears to be the executable for the service on the server side, but Where do they come from?  Where are they compiled?  Is the code available for those files? How can a person verify what they actually are?  

Not trying to be a pest, just can't seem to find any information about these files and where they come from.  

Thanks,

**Comment by wh1te909** [comment]
code is here https://github.com/amidaware/tacticalrmm/blob/develop/main.go

**Comment by SimpleTechGuy** [comment]
Thanks for the awesome fast response, but I guess I'm still a little confused.  Maybe it's because I haven't wrapped my mind entirely around nats, or even go for that matter, but I just can't figure out why the service binary's need to be pre-built rather than compiled with the go build command during install.

**Reply by wh1te909** [reply]
makes deployment easier cuz don't need to manage a go toolchain on each instance, one less thing to deal with. I am compiling them myself now locally but haven't had time yet to add a github action for it so that they're compiled on release, need to set that up

**Reply by SimpleTechGuy** [reply]
Roger that!  Good enough for me.  Thank you!
